crates/sui-framework/packages/deepbook/sources/clob_v3.move

```diff
@@ -0,0 +1,3043 @@
+        /// ID of the order defined by client
+        client_order_id: u128,
+        is_bid: bool,
+        /// owner ID of the `AccountCap` that canceled the order
+        owner: address,
+        original_quantity: u64,
+        base_asset_quantity_canceled: u64,
+        price: u64
+    }
+
+    /// A struct to make all orders canceled a more effifient struct
+    struct AllOrdersCanceledComponent<phantom BaseAsset, phantom QuoteAsset> has copy, store, drop {
+        /// ID of the order within the pool
+        order_id: u128,
+        /// ID of the order defined by client
+        client_order_id: u128,
+        is_bid: bool,
+        /// owner ID of the `AccountCap` that canceled the order
+        owner: address,
+        original_quantity: u64,
+        base_asset_quantity_canceled: u64,
+        price: u64
+    }
+
+    /// Emitted when batch of orders are canceled.
+    struct AllOrdersCanceled<phantom BaseAsset, phantom QuoteAsset> has copy, store, drop {
+        /// object ID of the pool the order was placed on
+        pool_id: ID,
+        orders_canceled: vector<AllOrdersCanceledComponent<BaseAsset, QuoteAsset>>,
+    }
+
+    /// Emitted only when a maker order is filled.
+    struct OrderFilled<phantom BaseAsset, phantom QuoteAsset> has copy, store, drop {
+        /// object ID of the pool the order was placed on
+        pool_id: ID,
+        /// ID of the order within the pool
+        order_id: u128,
+        /// ID of the order defined by taker client
+        taker_client_order_id: u128,
+        /// ID of the order defined by maker client
+        maker_client_order_id: u128,
+        is_bid: bool,
+        /// owner ID of the `AccountCap` that filled the order
+        taker_address: address,
+        /// owner ID of the `AccountCap` that placed the order
+        maker_address: address,
+        original_quantity: u64,
+        base_asset_quantity_filled: u64,
+        base_asset_quantity_remaining: u64,
+        price: u64,
+        taker_commission: u64,
+        maker_rebates: u64
+    }
+
+    /// Emitted when user deposit asset to custodian
+    struct DepositAsset<phantom Asset> has copy, store, drop {
+        /// object id of the pool that asset deposit to
+        pool_id: ID,
+        /// quantity of the asset deposited
+        quantity: u64,
+        /// owner address of the `AccountCap` that deposit the asset
+        owner: address
+    }
+
+    /// Emitted when user withdraw asset from custodian
+    struct WithdrawAsset<phantom Asset> has copy, store, drop {
+        /// object id of the pool that asset withdraw from
+        pool_id: ID,
+        /// quantity of the asset user withdrew
+        quantity: u64,
+        /// owner ID of the `AccountCap` that withdrew the asset
+        owner: address
+    }
+    // <<<<<<<<<<<<<<<<<<<<<<<< Events <<<<<<<<<<<<<<<<<<<<<<<<
+
+    struct Order has store, drop {
+        // For each pool, order id is incremental and unique for each opening order.
```

Comment:
nit: This comment now needs to be updated for the new Order ID (athough we also talked about maybe using a 256bit `ID` or `address` for this, right?)